api/service.py

```
@@ -421,6 +429,45 @@ async def _process_telemetry(telemetry: TelemetryInput, request_start: float) ->
+                active_connections=telemetry.active_connections or 0,
+                failure_occurred=is_anomaly
+            )
+
+            # Add training data
+            await predictive_engine.add_training_data(ts_data)
+
+            # Check for failure predictions
+            predictions = await predictive_engine.predict_failures(ts_data)
+
+            if predictions:
+                logger.info(f"Predictive maintenance: {len(predictions)} failure predictions made")
```

Ruff (0.14.10)

Undefined name `logger`

(F821)